src/blocking_client.rs

```diff
@@ -0,0 +1,41 @@
+use std::path::PathBuf;
+use tokio::runtime::{Builder, Runtime};
+use tonic_lnd::signrpc::{KeyDescriptor, KeyLocator, SignMessageReq, SignMessageResp};
+use tonic_lnd::{connect, Client};
+
+// For some LND grpc calls, we need a blocking version of the client. Namely, in order to use LDK's "sign"
+// API for signing an invoice request, the provided closure containing the LND signing API calls can't be
+// asynchronous.
+pub(crate) struct BlockingClient {
```

Comment:
Hm, can we implement these on the existing LND client then just have a small shim `BlockingClient` that calls `block_on(async client call)`. That way all of the LND logic still lives in one place. 

Also can we call this `SynchronousClient`?

Reply:
Yeah I think that would work! So I'm imagining like an lndclient struct that wraps the normal tonic lnd struct. Then also holds the BlockingClient as a field for this particular use case... LMK if that sounds like sort of what you're thinking of.